### PR TITLE
Docstrings and edit profile

### PR DESCRIPTION
**Pull request by 1sebba1**
Added loads of docstrings and made a basic edit email and password page on the edit profile screen. Also added an empty edit password method to the Auth util so the user can change their password. this is TBI.

### CODE REVIEW

File: lib/screens/home_page.dart [hunk 1]
```diff
@@ -107,17 +120,7 @@ class _HomePageState extends State<HomePage> {
     return filteredCourses;
   }
 
-  List<Course> filterCoursesByIterests(List interests, List<Course> courses) {
```
Comment (Akomea) [inline lib/screens/home_page.dart]: why did you remove this?

File: lib/screens/home_page.dart [hunk 2]
```diff
@@ -107,6 +107,17 @@ class _HomePageState extends State<HomePage> {
     return filteredCourses;
   }
 
+  List<Course> filterCoursesByIterests(List interests, List<Course> courses) {
```
Comment (Akomea) [inline lib/screens/home_page.dart]: ah it's back

File: lib/screens/edit_user_profile_page.dart [hunk 3]
```diff
@@ -14,12 +16,80 @@ class EditUserProfilePage extends StatefulWidget {
...
+  void initState() {
+    userNotifier = Provider.of<UserNotifier>(context, listen: false);
+    _nameController = TextEditingController();
+    _emailController = TextEditingController();
+    super.initState();
+    // _passwordController = TextEditingController();
+  }
+
   Widget _title() {
-    return Text(
```
Comment (Akomea) [inline lib/screens/edit_user_profile_page.dart:16]: Let's keep this to be consistent with the app design

**Review by Akomea — APPROVED** [review]
Good job, let's focus on documentation going forward

File: lib/screens/home_page.dart (hunk 1 above)
Comment (1sebba1) [reply lib/screens/home_page.dart]: It its a duplicate method, misspelt and never used

File: lib/screens/home_page.dart (hunk 2 above)
Comment (1sebba1) [reply lib/screens/home_page.dart]: look in the class you'll understand there is an identical method

File: lib/screens/edit_user_profile_page.dart (hunk 3 above)
Comment (1sebba1) [reply lib/screens/edit_user_profile_page.dart:16]: Keep the title?

Comment (1sebba1) [reply lib/screens/edit_user_profile_page.dart:16]: I put the style back, if that's what you meant

**Review by Akomea — APPROVED** [review]
